unix/write_stubs.c

```diff
@@ -0,0 +1,69 @@
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/custom.h>
+#include <caml/callback.h>
+#include <caml/alloc.h>
+#include <caml/unixsupport.h>
+#include <caml/bigarray.h>
+#include <caml/threads.h>
+
+#include <stdio.h>
+#include <errno.h>
+
+CAMLprim value stub_cstruct_write(value val_fd, value val_c)
+{
+  CAMLparam2(val_fd, val_c);
+  CAMLlocal3(val_buf, val_ofs, val_len);
+  val_buf = Field(val_c, 0);
+  val_ofs = Field(val_c, 1);
+  val_len = Field(val_c, 2);
+  void *buf = (char *)Caml_ba_data_val(val_buf) + Long_val(val_ofs);
```

Comment:
```suggestion
  uint8_t *buf = (uint8_t *)Caml_ba_data_val(val_buf) + Long_val(val_ofs);
```

Reply:
I approve this change, the FFI about `bigarray` says that it's an `uint8_t` array. We should keep this assumption on the C side too.